lib/database/leaderboard.php

```diff
@@ -400,19 +400,21 @@ function GetLeaderboardEntriesDataJSON($lbID, $user, $numToFetch, $offset, $frie
     return $retVal;
 }
 
-function GetLeaderboardData($lbID, $user, $numToFetch, $offset, $friendsOnly)
+function GetLeaderboardData($lbID, $user, $numToFetch, $offset, $friendsOnly, $nearby = 0)
 {
     sanitize_sql_inputs($lbID, $user, $numToFetch, $offset);
 
     $retVal = [];
 
     //    Get raw LB data
-    $query = "SELECT ld.ID AS LBID, gd.ID AS GameID, gd.Title AS GameTitle, ld.LowerIsBetter, ld.Title AS LBTitle, ld.Description AS LBDesc, ld.Format AS LBFormat, ld.Mem AS LBMem, ld.Author AS LBAuthor, gd.ConsoleID, c.Name AS ConsoleName, gd.ForumTopicID, gd.ImageIcon AS GameIcon,
-        ld.Author AS LBAuthor, ld.Created AS LBCreated, ld.Updated AS LBUpdated
+    $query = "SELECT ld.ID AS LBID, gd.ID AS GameID, gd.Title AS GameTitle, ld.LowerIsBetter, ld.Title AS LBTitle, ld.Description AS LBDesc, ld.Format AS LBFormat, ld.Mem AS LBMem, ld.Author AS LBAuthor, gd.ConsoleID, c.Name AS ConsoleName, gd.ForumTopicID, gd.ImageIcon AS GameIcon, ld.Author AS LBAuthor, ld.Created AS LBCreated, ld.Updated AS LBUpdated, COUNT(le.UserID) AS TotalEntries
               FROM LeaderboardDef AS ld
               LEFT JOIN GameData AS gd ON gd.ID = ld.GameID
               LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-              WHERE ld.ID = $lbID ";
+              LEFT JOIN leaderboardentry le ON le.LeaderboardID = ld.ID
+              LEFT JOIN UserAccounts AS ua ON ua.ID = le.UserID
+              WHERE ld.ID = $lbID
+              AND !ua.Untracked";
```

Comment:
i'm not an sql expert, but wouldn't it be more efficient to issue a separate query for the total entries instead of trying to join those records into the existing query?

Reply:
I had that originally but ended up going with this because it looked cleaner and I didn't think the performance would be that different. I'll do some tests on timing to see if the difference is noticeable.

Reply:
Alright so I finished testing this as well now.
The combined query I currently have seems to be a small bit faster than the alternative of two separate queries
`SELECT COUNT(le.UserID) AS TotalEntries
              FROM LeaderboardDef AS ld
              LEFT JOIN leaderboardentry le ON le.LeaderboardID = ld.ID
              LEFT JOIN UserAccounts AS ua ON ua.ID = le.UserID
              WHERE ld.ID = 92
              AND !ua.Untracked`
and
`SELECT ld.ID AS LBID, gd.ID AS GameID, gd.Title AS GameTitle, ld.LowerIsBetter, ld.Title AS LBTitle, ld.Description AS LBDesc, ld.Format AS LBFormat, ld.Mem AS LBMem, ld.Author AS LBAuthor, gd.ConsoleID, c.Name AS ConsoleName, gd.ForumTopicID, gd.ImageIcon AS GameIcon, ld.Author AS LBAuthor, ld.Created AS LBCreated, ld.Updated AS LBUpdated
              FROM LeaderboardDef AS ld
              LEFT JOIN GameData AS gd ON gd.ID = ld.GameID
              LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
              WHERE ld.ID = $lbID`

based on a timer in the php code starting before and after the queries are finished. Very small pool of results from a leaderboard with 5400 entries

Non-Cached Results
Combined  - Separate
0.00940  ---   0.01145
0.00932  ---  0.00946
0.00929  ---  0.00960

Cached Results
Combined - Separate
0.00012 --- 0.00019
0.00010 --- 0.00017
0.00019 --- 0.00017